### PR TITLE
use OpenAiEmbeddings and langchains to support large file

### PR DESCRIPTION
**Pull request by romaissaMe**
@achab 
ps: I removed anonymization in this Pr too cuz I forgot to merge the Pr of  feat/ remove-anonymization last time and I forgot about it  :)

### CODE REVIEW

File: app/utils/large_files.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+from langchain.vectorstores import FAISS 
+from langchain.chains.question_answering import load_qa_chain
+from langchain.llms import OpenAI
+from langchain import PromptTemplate 
+
+load_dotenv()
+
+langchain.embeddings.openai.api_key = os.getenv("OPENAI_API_KEY")
+
+def useEmbeddings(text: str,question:str):
```
Comment (achab) [inline app/utils/large_files.py:15]: can you name it `use_embeddings`? in python, we use snake_case convention

File: requirements.txt [hunk 2]
```diff
@@ -9,4 +9,7 @@ fastapi==0.97.0
 uvicorn==0.22.0
 python-dotenv==1.0.0
 requests~=2.31.0
-https://github.com/explosion/spacy-models/releases/download/en_core_web_lg-3.5.0/en_core_web_lg-3.5.0.tar.gz
\ No newline at end of file
+https://github.com/explosion/spacy-models/releases/download/en_core_web_lg-3.5.0/en_core_web_lg-3.5.0.tar.gz
+langchain==0.0.209
+tiktoken==0.4.0
+faiss-cpu==1.7.4
```
Comment (achab) [inline requirements.txt:15]: what does this library do?

File: app/utils/large_files.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    # Splitting up the text into smaller chunks for indexing
+    text_splitter = CharacterTextSplitter(        
+        separator = "\n",
+        chunk_size = 1000,
+        chunk_overlap  = 100, #striding over the text
+        length_function = len,
+    )
+    texts = text_splitter.split_text(text)
+    #create and store embeddings for each chunck
```
Comment (achab) [inline app/utils/large_files.py:24]: where are stored the embeddings?

File: app/utils/large_files.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    #define customized prompt
+    prompt = PromptTemplate(input_variables=['context', 'question'], 
+                            output_parser=None, partial_variables={}, template="I want you to play Paralegal Ai expert. you receive anonymized legal documents from both attorneys and non-legislatives. You are the legal Ai assistant, you have to analyze them,understand every word and sentence, explain what is written there, extract complex concepts, and serve as a guide for beginners and experts alike. You will be asked questions about the document, your mission is to answer them with the highest accuracy possible, you must work through it by heart to deliver correct knowledge and information extracted from the document which align perfectly with the user enquiries about the file and to develop useful resources from your current and previous experiences that user can use and will be beneficial to him.Use the following pieces of context to answer the question at the end. If you don't know the answer, just say that you don't know, don't try to make up an answer.\n\n{context}\n\nQuestion: {question}\nHelpful Answer:", template_format='f-string', validate_template=True)
```
Comment (achab) [inline app/utils/large_files.py:30]: can you store the template string in a variable elsewhere?

File: requirements.txt (hunk 2 above)
Comment (romaissaMe) [reply requirements.txt:15]: it's library to handle vectors and similarity search , tiktoken and fais-cpu are required for langchian library to work properly

File: app/utils/large_files.py (hunk 3 above)
Comment (romaissaMe) [reply app/utils/large_files.py:24]: docsearch = FAISS.from_texts(texts, embeddings)
stored in docsearch , faiss utilize OpenAiEmbeddings to create vector representations for texts

Comment (achab) [reply app/utils/large_files.py:25]: ok so it's not stored in a DB. just into a variable

**Pull request merged**